idaes/models_extra/co2_capture_and_utilization/unit_models/membrane_1d.py

```diff
@@ -0,0 +1,341 @@
+        ),
+    )
+
+    Stream_Config.declare(
+        "has_pressure_balance",
+        ConfigValue(
+            default=True,
+            domain=Bool,
+            doc="Bool indicating whether to include pressure balance for stream. Default=True.",
+        ),
+    )
+
+    Stream_Config.declare(
+        "has_feed",
+        ConfigValue(
+            default=True,
+            domain=Bool,
+            doc="Bool indicating whether stream has a feed.",
+            description="Bool indicating whether stream has a feed Port and inlet "
+            "state, or if all flow is provided via mass transfer. Default=True.",
+        ),
+    )
+    CONFIG.declare(
+        "sweep_flow",
+        ConfigValue(
+            default=True,
+            domain=Bool,
+            doc="Bool indicating whether there is a sweep flow in the permeate side.",
+            description="Bool indicating whether stream has a feed Port and inlet "
+            "state, or if all flow is provided via mass transfer. Default=True.",
+        ),
+    )
+
+    CONFIG.declare(
+        "finite_elements",
+        ConfigValue(
+            default=5,
+            domain=int,
+            description="Number of finite elements in length domain",
+            doc="""Number of finite elements to use when discretizing length
+                domain (default=5)""",
+        ),
+    )
+
+    CONFIG.declare(
+        "flow_type",
+        ConfigValue(
+            default=MembraneFlowPattern.countercurrent,
+            domain=In(MembraneFlowPattern),
+            description="Flow configuration of membrane",
+            doc="""Flow configuration of membrane
+               - MembraneFlowPattern.cocurrent: feed and sweep flows from 0 to 1
+               - MembraneFlowPattern.countercurrent: feed side flows from 0 to 1
+                                                    sweep side flows from 1 to 0  (default)""",
+        ),
+    )
+
+    CONFIG.declare(
+        "property_package",
+        ConfigValue(
+            default=None,
+            domain=is_physical_parameter_block,
+            description="Property package to use for control volume",
+            doc="""Property parameter object used to define property
+            calculations
+            (default = 'use_parent_value')
+            - 'use_parent_value' - get package from parent (default = None)
+            - a ParameterBlock object""",
+        ),
+    )
+
+    CONFIG.declare(
+        "property_package_args",
+        ConfigValue(
+            default={},
+            description="Arguments for constructing property package",
+            doc="""A dict of arguments to be passed to the PropertyBlockData
+            and used when constructing these
+            (default = 'use_parent_value')
+            - 'use_parent_value' - get package from parent (default = None)
+            - a dict (see property package for documentation)""",
+        ),
+    )
+
+    for side_name in ["feed", "sweep"]:
```

Comment:
Something to think about: "retentate" and "permeate" are more standard (and general) names for the sides of a membrane (especially for cases with no sweep stream).

Reply:
Thanks for the insight! May use them in the future if needed.